File: base-class-module/src/main/java/jingya/com/base_class_module/BaseViews/BasePopupWindow.java
package jingya.com.base_class_module.BaseViews;

import android.app.Activity;
import android.content.Context;
import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.graphics.drawable.ColorDrawable;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;
import android.widget.PopupWindow;

/**
 * @author Kuky
 */
public abstract class BasePopupWindow<VB extends ViewDataBinding> extends PopupWindow {
    protected Context mContext;
    protected VB mViewBinding;

    public BasePopupWindow(Context context, final Window window, int animStyle) {
        this(context, window, animStyle, 0);
    }

    public BasePopupWindow(Context context, final Window window, int animStyle, int width) {
        this(context, window, animStyle, width, 0);
    }

    public BasePopupWindow(Context context, final Window window, int animStyle, int width, int height) {
        super(context);
        init(context, window, animStyle, width, height);
    }

    private void init(Context context, final Window window, int animStyle, int width, int height) {
        this.mContext = context;
        this.mViewBinding = DataBindingUtil.inflate(LayoutInflater.from(mContext), getLayoutId(), null, false);
        setContentView(mViewBinding.getRoot());
        initPopupView();
        setFocusable(true);
        setAnimationStyle(animStyle);
        setBackgroundDrawable(new ColorDrawable(0));
        setWidth(width == 0 ? ViewGroup.LayoutParams.MATCH_PARENT : width);
        setHeight(height == 0 ? ViewGroup.LayoutParams.WRAP_CONTENT : height);
        setOnDismissListener(new OnDismissListener() {
            @Override
            public void onDismiss() {
                setBackgroundAlpha(1.0f);
            }
        });
    }

    public void setBackgroundAlpha(float alpha) {
        WindowManager.LayoutParams lp = ((Activity) mContext).getWindow().getAttributes();
        lp.alpha = alpha;
        ((Activity) mContext).getWindow().setAttributes(lp);
    }

    protected abstract int getLayoutId();

    protected abstract void initPopupView();
}

File: base-kt-module/src/main/java/com/kuky/base_kt_module/BaseViews/BasePopupWindow.kt
package com.kuky.baselib.baseClass

import android.app.Activity
import android.content.Context
import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.graphics.drawable.ColorDrawable
import android.view.LayoutInflater
import android.view.WindowManager
import android.widget.PopupWindow
import com.kuky.base_kt_module.R

/**
 * @author kuky
 * @description
 */
abstract class BasePopupWindow<VB : ViewDataBinding>(context: Context, animStyle: Int = R.style.animationForBottomAndBottom,
                                                     width: Int = WindowManager.LayoutParams.MATCH_PARENT,
                                                     height: Int = WindowManager.LayoutParams.WRAP_CONTENT) : PopupWindow(context) {
    protected var mContext: Context? = null
    protected lateinit var mViewBinding: VB

    init {
        initPopupWindow(context, animStyle, width, height)
    }

    private fun initPopupWindow(context: Context, animStyle: Int, width: Int, height: Int) {
        this.mContext = context
        this.mViewBinding = DataBindingUtil.inflate(LayoutInflater.from(mContext!!), getLayoutId(), null, false)
        contentView = mViewBinding.root
        initPopupView()
        setBackgroundDrawable(ColorDrawable(0))
        animationStyle = animStyle
        setWidth(width)
        setHeight(height)
        isFocusable = true
        setOnDismissListener({ setBackgroundAlpha(1.0f) })
    }

    fun setBackgroundAlpha(alpha: Float) {
        val lp = (mContext as Activity).window.attributes
        lp.alpha = alpha
        (mContext as Activity).window.attributes = lp
    }

    abstract fun getLayoutId(): Int

    fun initPopupView(){}
}